# Escenario 1: Opinión de clientes (Análisis de Sentimiento)

## 1. Perspectiva de negocio
La empresa de telecomunicaciones recibe cientos de comentarios diarios sobre su nuevo plan.  
El reto es transformar ese volumen de texto en **información accionable** para:
- Detectar problemas recurrentes (ej. mala señal).
- Identificar fortalezas (ej. atención al cliente, velocidad).
- Priorizar mejoras en producto y comunicación.

## 2. Interpretación esperada
- Comentarios positivos → polaridad > 0.
- Comentarios negativos → polaridad < 0.
- Comentarios mixtos/neutros → polaridad ≈ 0.

## 3. Valor de negocio
- **Marketing**: resalta atributos valorados (precio, atención, velocidad).  
- **Producto/operaciones**: prioriza mejoras en la señal, principal fuente de insatisfacción.  
- **Negocio**: permite reaccionar rápido, mejorar percepción de marca y reducir churn.

## 4. Nota técnica
TextBlob funciona mejor en inglés.  
Por eso las opiniones se traducen o se escriben directamente en inglés para obtener polaridades distintas de cero.


In [2]:
from textblob import TextBlob

opiniones_en = [
    "The new plan is excellent and very affordable",
    "The signal is bad and the service is slow",
    "It's not bad, but I expected something better",
    "Very satisfied with customer service",
    "The plan is cheap but the signal drops a lot",
    "The internet speed is amazing"
]

for texto in opiniones_en:
    polaridad = TextBlob(texto).sentiment.polarity
    print(f"Text: {texto}")
    print(f"Polarity: {polaridad:.2f}\n")


Text: The new plan is excellent and very affordable
Polarity: 0.45

Text: The signal is bad and the service is slow
Polarity: -0.50

Text: It's not bad, but I expected something better
Polarity: 0.25

Text: Very satisfied with customer service
Polarity: 0.65

Text: The plan is cheap but the signal drops a lot
Polarity: 0.40

Text: The internet speed is amazing
Polarity: 0.60



## Conclusiones del resultado

### 1. Opiniones positivas
- **0.45** → "The new plan is excellent and very affordable"  
- **0.65** → "Very satisfied with customer service"  
- **0.60** → "The internet speed is amazing"  

👉 Los clientes valoran el **precio accesible**, la **atención al cliente** y la **velocidad de internet**.

### 2. Opiniones negativas
- **-0.50** → "The signal is bad and the service is slow"  

👉 La **señal y el servicio lento** son los principales puntos de insatisfacción.

### 3. Opiniones mixtas / neutrales
- **0.25** → "It's not bad, but I expected something better"  
- **0.40** → "The plan is cheap but the signal drops a lot"  

👉 Comentarios con percepciones encontradas: el **precio es positivo**, pero la **calidad de la señal** genera frustración.

---

## Insight general
- **Fortalezas**: precio, atención al cliente, velocidad de internet.  
- **Debilidad crítica**: señal y estabilidad del servicio.  
- **Opiniones mixtas**: muestran que los clientes valoran algunos aspectos, pero esperan mejoras técnicas.

---

## Valor para el negocio
- **Marketing**: destacar atributos positivos en campañas (precio, velocidad, atención).  
- **Producto/operaciones**: priorizar mejoras en la señal, principal fuente de quejas.  
- **Estrategia**: el análisis de sentimiento permite reaccionar rápido, mejorar la percepción de marca y reducir la pérdida de clientes.